### PR TITLE
Add source code of RAVL tree

### PR DESCRIPTION
**Pull request by ldorau**

### CODE REVIEW

File: src/ravl/ravl.c [hunk 1]
```diff
@@ -0,0 +1,542 @@
+// SPDX-License-Identifier: BSD-3-Clause
```
Comment (bratpiorka) [inline src/ravl/ravl.c:1]: please update license and copyright date

File: src/ravl/ravl.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright 2018-2021, Intel Corporation */
```
Comment (bratpiorka) [inline src/ravl/ravl.h:2]: please update the license and date

File: src/ravl/ravl.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+/*
+ * ravl.h -- internal definitions for ravl tree
+ */
+
+#ifndef LIBPMEMOBJ_RAVL_H
```
Comment (bratpiorka) [inline src/ravl/ravl.h:8]: please update include guards

**Review by bratpiorka — CHANGES_REQUESTED** [review]

File: src/ravl/ravl.c (hunk 1 above)
Comment (ldorau) [reply src/ravl/ravl.c]: Done

File: src/ravl/ravl.h (hunk 3 above)
Comment (ldorau) [reply src/ravl/ravl.h]: Done

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**